torch/_subclasses/meta_utils.py

```diff
@@ -262,6 +262,153 @@ def sym_sizes_strides_storage_offset(
             self.check_for_expired_weak_storages()
             self.check_expired_count = 0
 
+        def empty_create(
+            inner_t,
+            inner_src,
+            inner_policy=policy,
+        ):
+            (
+                inner_sizes,
+                inner_strides,
+                inner_storage_offset,
+            ) = sym_sizes_strides_storage_offset(
+                inner_t,
+                inner_src,
+                inner_policy=inner_policy,
+            )
+            return torch.empty_strided(
+                inner_sizes,
+                inner_strides,
+                dtype=inner_t.dtype,
+                device="meta",
+            )
```

Comment:
Moved here from below and added an `inner_policy` arg for overriding the outer policy if needed

Reply:
As a stylistic note, these functions are relatively short so it doesn't matter that much, but since you DON'T ever need to access, e.g., policy, from inside the function bodies, it's better to not prepend inner_ and have the variables shadow the outers so it's impossible to access the wrong ones.